app/services/search-service.js

```diff
@@ -289,13 +289,24 @@ import './solr';
                 this.localizeFields = function localizeFields(fields){
                     if(!fields)
                         return;
-                        
-                    if(locale && locale!='en'){
-                        return fields.replace(/_EN/ig, '_'+(locale||'en').toUpperCase())
-                    }
 
+                    if (locale && locale !== 'en') {
+                        if (typeof fields === 'string') {
+                            return fields.replace(/_EN/ig, '_'+(locale||'en').toUpperCase())
+                        }
+                
+                        if (Array.isArray(fields)) {
+                            return fields.map(field => 
+                                typeof field === 'string' 
+                                    ? field.replace(/_EN/ig, '_'+(locale||'en').toUpperCase())
+                                    : field
+                            );
+                        }
+                    }
```

Comment:
safer to use `_EN_` instead of `_EN` and without /i.
I simplified the code also
```suggestion
                    if (locale && locale !== 'en') {
                        const fieldLocale = (locale||'en').toUpperCase();
                        const localize = (f) => typeof fields !== 'string' ? f.replace(/_EN_/g, `_${ fieldLocale }_`) : f;

                        if (Array.isArray(fields)) {
                            return fields.map(localize);
                        }

                        return localize(fields);
                    }
```